cudax/test/launch/launch_smoke.cu

```diff
@@ -0,0 +1,239 @@
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+#define LIBCUDACXX_ENABLE_EXCEPTIONS
+#include <cuda/atomic>
+#include <cuda/experimental/launch.cuh>
+
+#include <functional>
+#include <iostream>
+#include <type_traits>
+
+#include "../hierarchy/testing_common.cuh"
+
+__managed__ bool kernel_run_proof = false;
+
+void check_kernel_run(cudaStream_t stream)
+{
+  CUDART(cudaStreamSynchronize(stream));
+  CHECK(kernel_run_proof);
+  kernel_run_proof = false;
+}
+
+struct functor_int_argument
+{
+  __device__ void operator()(int dummy)
+  {
+    kernel_run_proof = true;
+  }
+};
+
+template <unsigned int BlockSize>
+struct functor_taking_config
+{
+  template <typename Config>
+  __device__ void operator()(Config conf, int grid_size)
+  {
+    static_assert(conf.dims.static_count(cudax::thread, cudax::block) == BlockSize);
+    assert(conf.dims.count(cudax::block, cudax::grid) == grid_size);
+    kernel_run_proof = true;
+  }
+};
+
+template <unsigned int BlockSize>
+struct functor_taking_dims
+{
+  template <typename Dimensions>
+  __device__ void operator()(Dimensions dims, int grid_size)
+  {
+    static_assert(dims.static_count(cudax::thread, cudax::block) == BlockSize);
+    assert(dims.count(cudax::block, cudax::grid) == grid_size);
+    kernel_run_proof = true;
+  }
+};
+
+__global__ void kernel_int_argument(int dummy)
+{
+  kernel_run_proof = true;
+}
+
+template <typename Config, unsigned int BlockSize>
+__global__ void kernel_taking_config(Config conf, int grid_size)
+{
+  functor_taking_config<BlockSize>()(conf, grid_size);
+}
+
+template <typename Dims, unsigned int BlockSize>
+__global__ void kernel_taking_dims(Dims dims, int grid_size)
+{
+  functor_taking_dims<BlockSize>()(dims, grid_size);
+}
+
+struct my_dynamic_smem_t
+{
+  int i;
+};
+
+template <typename SmemType>
+struct dynamic_smem_single
+{
+  template <typename Config>
+  __device__ void operator()(Config conf)
+  {
+    auto& dynamic_smem = cudax::dynamic_smem_ref(conf);
+    static_assert(::cuda::std::is_same_v<SmemType&, decltype(dynamic_smem)>);
+    assert(__isShared(&dynamic_smem));
+    kernel_run_proof = true;
+  }
+};
+
+template <typename SmemType, size_t Extent>
+struct dynamic_smem_span
+{
+  template <typename Config>
+  __device__ void operator()(Config conf, int size)
+  {
+    auto dynamic_smem = cudax::dynamic_smem_span(conf);
+    static_assert(decltype(dynamic_smem)::extent == Extent);
+    static_assert(::cuda::std::is_same_v<SmemType&, decltype(dynamic_smem[1])>);
+    assert(dynamic_smem.size() == size);
+    assert(__isShared(&dynamic_smem[1]));
+    kernel_run_proof = true;
+  }
+};
+
+TEST_CASE("Smoke", "[launch]")
```

Comment:
To fix the CI failures about `__device__` lambdas and linkage, move the body of the test cases into a separate function, and define the test cases like:
```c++
TEST_CASE("Smoke", "[launch]")
{
  launch_smoke_test();
}
```